server/routes/api/documents/documents.ts

```diff
@@ -1136,10 +1159,21 @@ router.post(
     });
     authorize(user, "move", document);
 
-    const collection = await Collection.scope({
-      method: ["withMembership", user.id],
-    }).findByPk(collectionId, { transaction });
-    authorize(user, "updateDocument", collection);
+    invariant(
+      collectionId || document.template,
+      "collectionId is required to move a document"
+    );
+
+    if (collectionId) {
+      const collection = await Collection.scope({
+        method: ["withMembership", user.id],
+      }).findByPk(collectionId, { transaction });
+      authorize(user, "updateDocument", collection);
+    }
+
+    if (document.isWorkspaceTemplate) {
```

Comment:
I don't like these separate `if` statements, too easy for things to fall between the cracks. Can we change it to something like

```
if (collectionId) {
 // ...
} else if (document.template) {
 // ...
} else {
  throw
}
```